tutorials/scripting/gdextension/gdextension_c_example.rst

```diff
@@ -0,0 +1,2154 @@
+pointer for the operator.
+
+With this handy, we can easily create our comparison function in the same file:
+
+.. code-block:: c
+
+    bool is_string_name_equal(GDExtensionConstStringNamePtr p_a, const char *p_b)
+    {
+        // Create a StringName for the C string.
+        StringName string_name;
+        constructors.string_name_new_with_utf8_chars(&string_name, p_b);
+
+        // Compare both StringNames.
+        bool is_equal = false;
+        operators.string_name_equal(p_a, &string_name, &is_equal);
+
+        // Destroy the created StringName.
+        destructors.string_name_destructor(&string_name);
+
+        // Return the result.
+        return is_equal;
+    }
+
+This function simply creates a ``StringName`` from the argument, compares with
+the other one using the operator function pointer, and returns the result. Note
+that the return value for the operator is passed as an out reference, this is a
+common thing in the API.
+
+Let's go back the the ``gdexample.h`` file and add a couple of functions that
+will be used as the callbacks for the Godot API:
+
+.. code-block:: c
+
+    void *gdexample_get_virtual_with_data(void *p_class_userdata, GDExtensionConstStringNamePtr p_name);
+    void gdexample_call_virtual_with_data(GDExtensionClassInstancePtr p_instance, GDExtensionConstStringNamePtr p_name, void *p_virtual_call_userdata, const GDExtensionConstTypePtr *p_args, GDExtensionTypePtr r_ret);
+
+There are actually two ways of registering virtual methods. One only has the
+``get`` part, in which you give Godot a properly crafted function pointer which
+will be called. For this we would need to create another helper for each virtual
+method, something that is not very convenient. Instead, we use the second method
+which allows us to return any data, and then Godot will call a second callback
+and give us back this data along with the call information. We can simply give
+our own function pointer as custom data and then have a single callback for all
+virtual methods. Although in this example we will only use it for one method,
+this way is simpler to expand.
+
+So let's implement those two functions in the ``gdexample.c`` file:
+
+.. code-block:: c
+
+    void *gdexample_get_virtual_with_data(void *p_class_userdata, GDExtensionConstStringNamePtr p_name)
+    {
+        // If it is the "_process" method, return a pointer to the gdexample_process function.
+        if (is_string_name_equal(p_name, "_process"))
+        {
+            return (void *)gdexample_process;
+        }
+        // Otherwise, return NULL.
+        return NULL;
+    }
+
+    void gdexample_call_virtual_with_data(GDExtensionClassInstancePtr p_instance, GDExtensionConstStringNamePtr p_name, void *p_virtual_call_userdata, const GDExtensionConstTypePtr *p_args, GDExtensionTypePtr r_ret)
+    {
+        // If it is the "_process" method, call it with a helper.
+        if (is_string_name_equal(p_name, "_process"))
```

Comment:
This is re-creating the StringName `"_process"` every process frame, isn't it? It would be nice to avoid that here.